src/ceph-disk

```diff
@@ -2943,6 +2952,64 @@ def main_zap(args):
 
 ###########################
 
+def main_trigger(args):
+    if args.parttype in [OSD_UUID, MPATH_OSD_UUID]:
+        trigger_cmd('activate', args.dev)
+    elif args.partype in [JOURNAL_UUID, MPATH_JOURNAL_UUID]:
+        trigger_cmd('activate-journal', args.dev)
+
+    elif args.parttype in [DMCRYPT_JOURNAL_UUID]:
+        command(
+            [
+                '/sbin/cryptsetup',
+                '--key-file',
+                '/etc/ceph/dmcrypt-keys/{id} --key-size 256 create {id} {dev}'.format(
+                    id=args.partid,
+                    dev=args.dev,
+            ]
+        )
+    elif args.parttype in [DMCRYPT_LUKS_JOURNAL_UUID]:
+         command(
+            [
+                '/sbin/cryptsetup',
+                '--key-file',
+                '/etc/ceph/dmcrypt-keys/{id}.luks.key luksOpen {dev} {id}'.format(
+                    id=args.partid,
+                    dev=args.dev,
+            ]
+        )
+
+    else:
+        raise Error('unrecognized partition type %s', args.parttype)
```

Comment:
The current udev logic for dmcrypt looks more like:

* cryptsetup
* udevadm settle (well, it's sleep right now but maybe it should be that instead)
* ceph-disk activate